### ISSUE TITLE
limitToOptions is ignored by Paper UI

### ISSUE THREAD

**Issue by kaikreuzer** [eclipse/smarthome#2820]
I have a config parameter defined with provided options and limitToOptions=false:
```
      {  
         "name":"command",
         "type":"TEXT",
         "label":"Command",
         "description":"the command to be sent",
         "required":true,
         "limitToOptions":false,
         "options":[
            {
            "label": "ON",
            "value": "ON"
            },
            {
            "label": "OFF",
            "value": "OFF"
            },
            {
            "label": "OPEN",
            "value": "OPEN"
            },
            {
            "label": "CLOSED",
            "value": "CLOSED"
            },
            {
            "label": "UP",
            "value": "UP"
            },
            {
            "label": "DOWN",
            "value": "DOWN"
            }
          ]
      }
```
Nonetheless, I cannot enter any arbitrary value - I can only choose one of the given options:
<img width="288" alt="screen shot 2017-01-13 at 09 37 26" src="https://cloud.githubusercontent.com/assets/3244965/21923367/edc4bf1c-d973-11e6-8213-46c7d121a9be.png">

**Comment by aounhaider1** [comment]
Thats because its not supported here. It is only supported in multiple widget.

**Comment by kaikreuzer** [comment]
What do you mean? What is the difference of a "multiple widget" to a parameter with limitToOptions=false?

**Comment by aounhaider1** [comment]
For multiple=false we are using default material md-select. It does not provide the option to add new values. On the other hand, the widget that we are using for multiple=true is our own custom widget, that supports adding new values and so on.

**Comment by kaikreuzer** [comment]
So the bug is that the wrong widget type is used here? So I guess you can fix this.

**Comment by kaikreuzer** [comment]
Sorry, there are still issues with that widget after #2841.

1. Set values (which are not in the option list) are not shown as the current value when opening a page:
![screen shot 2017-01-18 at 09 51 51](https://cloud.githubusercontent.com/assets/3244965/22057171/d15836e4-dd63-11e6-93ee-2ecb5e1a5074.png)

On this picture, the parameter is set to "1", but it is not shown, unless you open the drop-down.

2. The field value is grey, if it is an option from the list and black, if it is a custom entry. Imho, on an limitToOptions=false field, the value should always be a String that the user can type and edit and thus it should always be shown in black.

**Comment by kaikreuzer** [comment]
Sorry, I am still not happy with this. It feels very unintuitive for the user when entering a custom value.

I would expect, he can enter a value and click "OK" to save it:
<img width="524" alt="screen shot 2017-01-18 at 17 11 22" src="https://cloud.githubusercontent.com/assets/3244965/22072051/5a773e56-dda1-11e6-85ba-ddda0aca5d48.png">
Unfortunately, this value is NOT stored - it is only stored, if the user clicks the green "+" first. I think this is a trap that many users will get caught in. The expectation is that this is a "normal" text input field, which additionally gives some pre-defined options to choose from.

Some additional problems that can be seen on this screenshot:
<img width="522" alt="screen shot 2017-01-18 at 17 14 07" src="https://cloud.githubusercontent.com/assets/3244965/22072181/d3ae14d4-dda1-11e6-8b39-153a2d2be214.png">
- The ordering is weird. "5" is listed above the empty entry, "10" below it.
- The drop down is cut off by modal dialogs. This makes it difficult to operate as it scrolls the drop-down and the dialog content at the same time.

**Comment by kaikreuzer** [comment]
Ok, these things are solved by #2866, thanks.

What is left is my comment on #2866: "Same actually applies to the check boxes on the right, they are rather confusing.". Is it too hard to remove them?
Another usability weirdness is that the drop-down does not automatically close after selecting a value. For multiple=true, it certainly makes sense to keep it open, for multiple=false imho not.

**Comment by aounhaider1** [comment]
Fixed.
